zerver/tests/test_events.py

```diff
@@ -3511,6 +3511,30 @@ def do_set_realm_permission_group_setting_test(self, setting_name: str) -> None:
             if user_group.name == default_group_name:
                 continue
 
+            if (
+                not setting_permission_configuration.allow_internet_group
+                and user_group.name == SystemGroups.EVERYONE_ON_INTERNET
+            ):
+                continue
+
+            if (
+                not setting_permission_configuration.allow_everyone_group
+                and user_group.name == SystemGroups.EVERYONE
+            ):
+                continue
+
+            if (
+                not setting_permission_configuration.allow_owners_group
+                and user_group.name == SystemGroups.OWNERS
+            ):
+                continue
+
+            if (
+                not setting_permission_configuration.allow_nobody_group
+                and user_group.name == SystemGroups.NOBODY
+            ):
```

Comment:
Similar code is used while checking if a setting can be set to a particular group, but we return different error for each case, so not possible to reuse that as of now.